scripts/shared/select/main.js

```diff
@@ -196,6 +214,68 @@ function makeDropdownAdapter(select) {
   return adapter;
 }
 
+/*
+ * Decorator for multiple select
+ * -- render selections in one line
+ * -- when the selections exceed the width of the selector, it gets clipped 
+ *    and instead the total number of selections are shown
+ */
+function MultipleSelectSingleLine() {}
+MultipleSelectSingleLine.prototype.update = function(decorated, data) {
+    this.clear();
+
+    if (data.length === 0) {
+      return;
+    }
+
+    var $selections_rendered = this.$selection.find('.select2-selection__rendered')
+    var available_width = $selections_rendered.width()
+    // the element to indicate how many total selections are currently selected
+    var selected_indicator = '<li class="select2-selection__choice__count">' + data.length + ' selected</li>'
+    $selections_rendered.append(selected_indicator)
+    selected_indicator = $('.select2-selection__choice__count')
+    var selected_indicator_width = selected_indicator.outerWidth(true)
+
+    var selected_indicator_added = false // flag so we only attach the selected_indicator element once
+    var selected_total_width = 0;
+
+    var $selections = [];
+
+    for (var d = 0; d < data.length; d++) {
+      var selection = data[d];
+
+      var formatted = this.display(selection);
+      var $selection = this.selectionContainer();
+
+      $selection.append(formatted);
+      $selection.data('data', selection);
+      
+      selection_width = measure_width($selection, $selections_rendered)
+      selected_total_width += selection_width
+      remaining_width = available_width - selected_total_width
+      if (remaining_width < selected_indicator_width) {
```

Comment:
I notice that this code doesn't work so well if you manually adjust the width of your browser.  Maybe you could get it to work more reliably with something like this:

```
.select2-selection__rendered {
  height: 31px;
  overflow: hidden;
}
```

I think you also would need to make sure the X selected div is exactly the same height as the chosen options.